### PR TITLE
fix: incorrect aggregation result of `bool_and`

### LINKED ISSUE #11846: Potential bug in `bool_and()` aggregate function
### Describe the bug

The last query in reproducer should return `true`

Run in datafusion-cli:
```
DataFusion CLI v40.0.0
> create table t1(v1 int, v2 boolean);
0 row(s) fetched.
Elapsed 0.073 seconds.

> insert into t1 values (1, true), (1, true);
+-------+
| count |
+-------+
| 2     |
+-------+
1 row(s) fetched.
Elapsed 0.051 seconds.

> select bool_and(v2) from t1;
+-----------------+
| bool_and(t1.v2) |
+-----------------+
| true            |
+-----------------+
1 row(s) fetched.
Elapsed 0.016 seconds.

> select v1, bool_and(v2) from t1 group by v1;
+----+-----------------+
| v1 | bool_and(t1.v2) |
+----+-----------------+
| 1  | false           |
+----+-----------------+
1 row(s) fetched.
Elapsed 0.011 seconds.
dd
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

Closes #11846.

## Rationale for this change
The initial value of each group was set to `false`. When encountering the first non-null value of a certain group and performing the AND operation with the initial value, it resulted in an incorrect output.

We should set the initial value to the identity element.

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/bool_op.rs [hunk 1]
```diff
@@ -77,7 +82,9 @@ where
 
         if self.values.len() < total_num_groups {
             let new_groups = total_num_groups - self.values.len();
-            self.values.append_n(new_groups, Default::default());
+            // Fill with the identity element, so that when the first non-null value is encountered,
+            // it will combine with the identity and the result will be the first non-null value itself.
+            self.values.append_n(new_groups, self.identity);
```
Comment (alamb) [inline datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/bool_op.rs:87]: Nice -- this is similar to the default value for the PrimitiveOp accumulator too

**Review by alamb — APPROVED** [review]
😍 

Thank you @jonahgao 

cc @2010YOUY01

**Review by Dandandan — APPROVED** [review]
✅

**Comment by jonahgao** [comment]
Thanks for the review.

**Pull request merged**